src/prefect/tasks/census/census.py

```diff
@@ -0,0 +1,125 @@
+                CensusSyncTask constructor. Keyword argument.
+            - poll_status_every_n_seconds (int, optional): this task polls the Census API for the sync's
+                status. If provided, this value will override the default polling time of
+                60 seconds and it has a minimum wait time of 5 seconds. Keyword argument.
+
+        Returns:
+            - dict: dictionary of statistics returned by Census on the specified sync
+        """
+
+        if not api_trigger:
+            raise ValueError(
+                """Value for parameter `api_trigger` must be provided. See Census sync
+                                configuration page."""
+            )
+
+        pattern = r"https:\/\/bearer:secret-token:(.*)@app.getcensus.com\/api\/v1\/syncs\/(\d*)\/trigger"
+        url_pattern = re.compile(pattern)
+
+        # Making sure it is a valid api trigger.
+        confirmed_pattern = url_pattern.match(api_trigger)
+
+        if not confirmed_pattern:
+            raise ValueError(
+                """Invalid parameter for `api_trigger` please paste directly from the Census
+                                sync configuration page. This is CaSe SenSITiVe."""
+            )
+
+        secret, sync_id = confirmed_pattern.groups()
+        response = requests.post(api_trigger)
+
+        if response.status_code != 200:
+            error_string = f"Sent POST, failed with status code {response.status_code}: {response.text}."
+            raise ValueError(error_string)
+
+        sleep_time = max(MIN_WAIT_TIME, poll_status_every_n_seconds)
+
+        self.logger.debug(
+            f"Started Census sync {sync_id}, sleep time set to {sleep_time} seconds."
+        )
+
+        sync_run_id = response.json()["data"]["sync_run_id"]
+        sr_url = f"https://bearer:secret-token:{secret}@app.getcensus.com/api/v1/sync_runs/{sync_run_id}"
+        log_url = f"https://app.getcensus.com/syncs/{sync_id}/sync-history"
+
+        result = {}
+
+        start_time = time.time()
+        while True:
+            time.sleep(sleep_time)
+            status_response = requests.get(sr_url)
+            response_dict = status_response.json()
+            if status_response.status_code != 200 or "data" not in response_dict.keys():
+                raise ValueError(
+                    f"Getting status of sync failed, please visit Census Logs at {log_url} to see more."
+                )
+            result = response_dict["data"]
+            status = result["status"]
+            if status == "working":
+                self.logger.debug(
+                    f"Sync {sync_id} still running after {round(time.time()-start_time, 2)} seconds."
+                )
+                continue
+            break
+
+        self.logger.debug(
+            f"Sync {sync_id} has finished running after {round(time.time()-start_time, 2)} seconds."
+        )
+        self.logger.info(f"View details here: {log_url}.")
```

Comment:
should this be debug level too?

Reply:
I think this should be info actually because this that link takes the user to the UI to be able to drill into the invalid or failed records on the syncs. 

If in general though: info-level logs are not encouraged, I'm happy to turn that into a debug level here and then return that log url in the response dictionary as well so it's up to the data eng.

Reply:
got it, `info` sound appropriate then, thanks for the detail here!